I am able to manually test the issue and the fix as well. 

Steps taken : 
--- Create an Execution Environment with a bad image name : ```quay.io/agautam/badimg```
--- Create a container group , and associate the container group and EE with a Job template
--- Run that Job Template and look for errors on CLI / GUI 

Error on the AWX CLI : 
```
tools_awx_1 | awx-dispatcher stdout | 2021-05-21 12:49:13,624 ERROR    [9b15d5ee] awx.main.tasks job 21 (running) Exception occurred while running task
tools_awx_1 | awx-dispatcher stdout | Traceback (most recent call last):
tools_awx_1 | awx-dispatcher stdout |   File "/awx_devel/awx/main/tasks.py", line 1382, in run
tools_awx_1 | awx-dispatcher stdout |     res = receptor_job.run()
tools_awx_1 | awx-dispatcher stdout |   File "/awx_devel/awx/main/tasks.py", line 2921, in run
tools_awx_1 | awx-dispatcher stdout |     return self._run_internal(receptor_ctl)
tools_awx_1 | awx-dispatcher stdout |   File "/awx_devel/awx/main/tasks.py", line 2984, in _run_internal
tools_awx_1 | awx-dispatcher stdout |     raise RuntimeError(detail)
tools_awx_1 | awx-dispatcher stdout | RuntimeError: Error creating pod: timed out waiting for the condition, ErrImagePull
```


Error on the AWX UI : 

![image](https://user-images.githubusercontent.com/7301586/119140499-379ac680-ba12-11eb-924f-c1b4e03b11d5.png)


Thank you @rooftopcellist for Helping me understand a bit about EE.

--- a/awx/main/tasks.py
+++ b/awx/main/tasks.py
@@ -3017,6 +3017,7 @@ class AWXReceptorJob:
 
             receptor_params = {
                 "secret_kube_pod": spec_yaml,
+                "pod_pending_timeout": getattr(settings, 'AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT', "5m"),
             }
 
             if self.credential:

--- a/awx/settings/defaults.py
+++ b/awx/settings/defaults.py
@@ -70,9 +70,9 @@ IS_K8S = False
 
 RECEPTOR_RELEASE_WORK = True
 AWX_CONTAINER_GROUP_K8S_API_TIMEOUT = 10
-AWX_CONTAINER_GROUP_POD_LAUNCH_RETRIES = 100
-AWX_CONTAINER_GROUP_POD_LAUNCH_RETRY_DELAY = 5
 AWX_CONTAINER_GROUP_DEFAULT_NAMESPACE = os.getenv('MY_POD_NAMESPACE', 'default')
+# Timeout when waiting for pod to enter running state. If the pod is still in pending state , it will be terminated. Valid time units are "s", "m", "h". Example : "5m" , "10s".
+AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT = "5m"
 
 # Internationalization
 # https://docs.djangoproject.com/en/dev/topics/i18n/

